.travis/cover.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2012-8 Met Office.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+from subprocess import call
+
+
+def main():
+    call(
+        'rose test-battery --state=save -j 5 || (echo -e "\n\nRerunning Failed Tests...\n\n"; rose test-battery -v --state=save,failed -j 5)',
+        shell=True)
```

Comment:
To reduce the shell syntax and to return with the correct exit code, perhaps we can try (not tested):

```python
command = ['rose', 'test-battery', '-j', '5']
if call(command + ['--state=save']):
    # Non-zero return code
    import sys
    sys.stderr.write('\n\nRerunning Failed Tests...\n\n')
    # Exit with final return code
    sys.exit(call(command + ['--state=save,failed', '-v']))
```

Reply:
Great idea @matthewrmshin ! Copied your code as-is, first it failed, but then the next run (where I squashed commits) it re-ran the tests and it worked. Before the script execution would always report success, passing the build :disappointed: 

Going to fix the Cylc pull request with this neat approach. Thanks Matt!